en/lessons/basics/modules.md

```diff
@@ -1,316 +1,195 @@
+entities that are elided.
 
-```elixir
-defmodule Example do
-  require SuperMacros
+```haskell
+module Example where
 
-  SuperMacros.do_stuff
-end
+import Data.Maybe (fromMaybe)
 ```
 
-If we attempt to call a macro that is not yet loaded Elixir will raise an error.
-
-### `use`
-
-With the `use` macro we can enable another module to modify our current module's definition.
-When we call `use` in our code we're actually invoking the `__using__/1` callback defined by the provided module.
-The result of the `__using__/1` macro becomes part of our module's definition.
-To get a better understanding how this works let's look at a simple example:
-
-```elixir
-defmodule Hello do
-  defmacro __using__(_opts) do
-    quote do
-      def hello(name), do: "Hi, #{name}"
-    end
-  end
-end
-```
+Another technique is to qualify an import so that all use sites are tagged.
 
-Here we've created a `Hello` module that defines the `__using__/1` callback inside of which we define a `hello/1` function.
-Let's create a new module so we can try out our new code:
 
-```elixir
-defmodule Example do
-  use Hello
-end
-```
+```haskell
+module Example where
 
-If we try our code out in IEx we'll see that `hello/1` is available on the `Example` module:
+import qualified Data.Maybe
 
-```elixir
-iex> Example.hello("Sean")
-"Hi, Sean"
+myFromMaybe = Data.Maybe.fromMaybe
 ```
 
-Here we can see that `use` invoked the `__using__/1` callback on `Hello` which in turn added the resulting code to our module.
-Now that we've demonstrated a basic example let's update our code to look at how `__using__/1` supports options.
-We'll do this by adding a `greeting` option:
-
-```elixir
-defmodule Hello do
-  defmacro __using__(opts) do
-    greeting = Keyword.get(opts, :greeting, "Hi")
+Note, you can use qualifications even if a module is not imported as
+`qualified`. This highlights that qualification enforces a namespace, but we can
+always optionally declare the namespace. This is useful for resolving naming
+collisions. In this example I will use ghci to demonstrate a name collision error.
 
-    quote do
-      def hello(name), do: unquote(greeting) <> ", " <> name
-    end
-  end
-end
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> import Data.Map
+ghci> import Data.List
+ghci> :t map
+<interactive>:1:1: error:
+    Ambiguous occurrence ‘map’
+    It could refer to
+       either ‘Data.Map.map’,
+              imported from ‘Data.Map’
+              (and originally defined in ‘Data.Map.Internal’)
+           or ‘Prelude.map’,
+              imported from ‘Prelude’ (and originally defined in ‘GHC.Base’)
+ghci> :t Data.Map.map
+ghci> :t Data.Map.map
+Data.Map.map :: (a -> b) -> Map k a -> Map k b
+ghci> :t Data.List.map
+Data.List.map :: (a -> b) -> [a] -> [b]
 ```
 
-Let's update our `Example` module to include the newly created `greeting` option:
+If `Data.Maybe` is too long to type out, we can provide an alias for our import
```

Comment:
Just to show idiomatic usage, would it perhaps be better if something that is commonly imported qualified, such as `Text`, were used?